src/libcollections/string.rs

```diff
@@ -1072,3 +1128,61 @@ impl fmt::Write for String {
         Ok(())
     }
 }
+
+/// A draining iterator for `String`.
+#[unstable(feature = "collections_drain", reason = "recently added")]
+pub struct Drain<'a> {
+    string: *mut String,
+    /// Start of part to remove
+    start: usize,
+    /// Index of tail to preserve
+    tail_start: usize,
+    /// Current remaining range to remove
+    iter: Chars<'a>,
+    _marker: PhantomData<&'a mut String>,
```

Comment:
Oh hm interesting! I was under the impression that variance was the only thing as well, but I also think this is fine to keep as there's certainly no harm in doing so!